### PR TITLE
upgrade v8 to disable pointer compression, fix memleak

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: core/runtime/jsruntime.rs [hunk 1]
```diff
@@ -326,7 +326,7 @@ fn v8_init(
   #[repr(C, align(16))]
   struct IcuData([u8; 10541264]);
   static ICU_DATA: IcuData = IcuData(*include_bytes!("icudtl.dat"));
-  v8::icu::set_common_data_72(&ICU_DATA.0).unwrap();
+  v8::icu::set_common_data_73(&ICU_DATA.0).unwrap();
```
Comment (mmastrac) [inline core/runtime/jsruntime.rs:329]: I had to downgrade the ICU data as part of https://github.com/denoland/deno_core/pull/153 -- do we need to upgrade it again?

Comment (ry) [reply core/runtime/jsruntime.rs:329]: Yes, this is back to v8 11.7.439.6 which depends on ICU 73

Comment (mmastrac) [reply core/runtime/jsruntime.rs:329]: Sorry, I meant the `icudtl.dat` file itself -- the version pre-downgrade was a different size (10631872). I'm not sure if we need to include that file in this update.

If we need to update this file, it should be available in this commit here: https://github.com/mmastrac/deno_core/blob/fe80221e774fdfde8640571a4b80446fa542d6bb/core/runtime/icudtl.dat

Comment (bartlomieju) [reply core/runtime/jsruntime.rs:329]: Yes, we should include the other data file.

Comment (ry) [reply core/runtime/jsruntime.rs:329]: fixed and added a test

**Review by mmastrac — APPROVED** [review]
LGTM! Great test.

**Pull request merged**